src/coreclr/jit/codegencommon.cpp

```diff
@@ -1733,55 +1733,42 @@ void CodeGen::genGenerateMachineCode()
         printf(" for ");
 
 #if defined(TARGET_X86)
-        if (compiler->info.genCPU == CPU_X86)
+        if (compiler->canUseEvexEncoding())
         {
-            printf("generic X86 CPU");
+            printf("X86 with AVX512");
         }
-        else if (compiler->info.genCPU == CPU_X86_PENTIUM_4)
+        else if (compiler->canUseVexEncoding())
         {
-            printf("Pentium 4");
+            printf("X86 with AVX");
         }
-#elif defined(TARGET_AMD64)
-        if (compiler->info.genCPU == CPU_X64)
+        else
         {
-            if (compiler->canUseEvexEncoding())
-            {
-                printf("X64 CPU with AVX512");
-            }
-            else if (compiler->canUseVexEncoding())
-            {
-                printf("X64 CPU with AVX");
-            }
-            else
-            {
-                printf("X64 CPU with SSE2");
-            }
+            printf("X86 with SSE2");
         }
-#elif defined(TARGET_ARM)
-        if (compiler->info.genCPU == CPU_ARM)
+#elif defined(TARGET_AMD64)
+        if (compiler->canUseEvexEncoding())
         {
-            printf("generic ARM CPU");
+            printf("X64 with AVX512");
         }
-#elif defined(TARGET_ARM64)
-        if (compiler->info.genCPU == CPU_ARM64)
+        else if (compiler->canUseVexEncoding())
         {
-            printf("generic ARM64 CPU");
+            printf("X64 with AVX");
         }
-#elif defined(TARGET_LOONGARCH64)
-        if (compiler->info.genCPU == CPU_LOONGARCH64)
+        else
         {
-            printf("generic LOONGARCH64 CPU");
+            printf("X64 with SSE2");
         }
+#elif defined(TARGET_ARM)
+        printf("generic ARM");
+#elif defined(TARGET_ARM64)
+        printf("generic ARM64");
```

Comment:
Since we accelerate ARM64, is it worth clarifying `ARM64 with AdvSimd`, as we do with the baseline (SSE2) case on `x86/x64`?

Reply:
Alternatively, is it worth having the baseline `X86`/`X64` case be `generic X86` and `generic X64`?